File: frontend/src/api/history.ts
import { request } from './request';

export interface HistoryItem {
  id: number;
  query: string;
  answer_summary: string;
  model_used: string;
  provider: string;
  source_files: string[];
  created_at: string;
}

export interface HistoryListResponse {
  items: HistoryItem[];
  total: number;
}

export async function getHistory() {
  const { data } = await request.get<HistoryListResponse>('/history');
  return data;
}

File: backend/app/api/history.py
from sqlalchemy import select

from fastapi import APIRouter

from app.api.deps import CurrentUser, DbSession
from app.models.chat_history import ChatHistory
from app.schemas.history import HistoryItem, HistoryListResponse


router = APIRouter(prefix="/history", tags=["history"])


@router.get("", response_model=HistoryListResponse)
def list_history(
    db: DbSession,
    current_user: CurrentUser,
) -> HistoryListResponse:
    stmt = (
        select(ChatHistory)
        .where(ChatHistory.user_id == current_user.id)
        .order_by(ChatHistory.created_at.desc())
        .limit(100)
    )
    records = list(db.scalars(stmt).all())

    return HistoryListResponse(
        items=[
            HistoryItem(
                id=record.id,
                query=record.query,
                answer_summary=record.answer_summary,
                model_used=record.model_used,
                provider=record.provider,
                source_files=record.source_files_json,
                created_at=record.created_at,
            )
            for record in records
        ],
        total=len(records),
    )
